app/models/r_package.rb

```diff
@@ -30,19 +30,45 @@ def self.database
   def self.data_for(answers, domain_ids)
     raise 'No domains given' unless domain_ids.present?
 
-    initial_hash = normalized_shadowcat answers: [], domain: domain_ids
-
-    # Recalculate the estimate and variance for <index> answers,
-    # then recalculate it for <index + 1> answers with the previous estimate and variance,
-    # repeat until done for all answers
-    answers.each_with_index.inject(initial_hash) do |hash, (_, index)|
-      params = { answers: [answers.take(index + 1).to_h],
-                 estimate: hash[:estimate],
-                 variance: hash[:variance],
-                 domain: domain_ids }
-
-      normalized_shadowcat params
+    #TODO: Move this outer loop to screensmart-r
+    #      so that it can handle multiple domains
+    #      without this ugly loop
+    domain_results = domain_ids.map do |domain_id|
+      accumulator = normalized_shadowcat answers: [], domain: [domain_id]
+
+      answers_in_domain = answers.select do |question_id, _|
+        domain_id_for_question_id(question_id) == domain_id
+      end
+
+      # Recalculate the estimate and variance for <index> answers,
+      # then recalculate it for <index + 1> answers with the previous estimate and variance,
+      # repeat until done for all answers
+      answers_in_domain.each_with_index.inject(accumulator) do |hash, (_, index)|
+        params = { answers: [answers_in_domain.take(index + 1).to_h],
+                   estimate: hash[:estimate],
+                   variance: hash[:variance],
+                   domain: [domain_id] }
+
+        normalized_shadowcat params
+      end
+    end
+
+    { next_question_id: domain_results.find { |dr| !dr[:done] }.try(:[], :next_question_id),
+      done: domain_results.all? { |domain| domain[:done] },
+      domain_results: domain_ids.zip(domain_results).map do |domain_id, domain|
+        [domain_id, domain.slice(:estimate, :variance, :estimate_interpretation, :warning)]
+      end.to_h
+    }
+  end
```

Comment:
Ik vind dit een nogal heftige methode geworden, erg complex.

Misschien wel goed dat de interpretatie van de R response nu meer hier leeft, maar het is er eigenlijk niet leesbaarder op geworden.